File: app/src/main/java/ru/p3tr0vich/fuel/DividerItemDecorationPreferences.java
package ru.p3tr0vich.fuel;

import android.content.Context;
import android.support.v7.preference.PreferenceCategory;
import android.support.v7.preference.PreferenceGroupAdapter;
import android.support.v7.widget.RecyclerView;

public class DividerItemDecorationPreferences extends DividerItemDecorationBase {

    DividerItemDecorationPreferences(Context context) {
        super(context);
    }

    @Override
    public boolean shouldDrawDivider(RecyclerView parent, int childViewIndex) {
        RecyclerView.Adapter adapter = parent.getAdapter();

        assert adapter != null;

        return !(((PreferenceGroupAdapter) adapter).getItem(childViewIndex)
                instanceof PreferenceCategory);
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/DividerItemDecorationPreferences.kt
package ru.p3tr0vich.fuel

import android.annotation.SuppressLint
import android.content.Context
import androidx.preference.PreferenceCategory
import androidx.preference.PreferenceGroupAdapter
import androidx.recyclerview.widget.RecyclerView

class DividerItemDecorationPreferences(context: Context?) : DividerItemDecorationBase(context) {

    @SuppressLint("RestrictedApi")
    public override fun shouldDrawDivider(parent: RecyclerView, childViewIndex: Int): Boolean {
        return (parent.adapter as PreferenceGroupAdapter?)?.getItem(childViewIndex) !is PreferenceCategory
    }
}